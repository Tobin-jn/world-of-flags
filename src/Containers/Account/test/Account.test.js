import React from 'react';
import { shallow, mount } from 'enzyme';
import { Account, mapStateToProps, mapDispatchToProps } from '../index';

describe('Account', () => {
  let wrapper;
  let mockUser;
  let signOut;
  let mockSignOut

  beforeEach(() => {
    mockSignOut = jest.fn()
    mockUser = {
            id: 2,
            name: 'rtg',
            loggedIn: true,
            email: 'alex@turing.com'
           }
    wrapper = shallow(<Account 
                        user={mockUser} 
                        signOut={mockSignOut}
                      />)
  });

  it('should render like the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('changeRoute', () => {
    it('should invoke forceUpdate', () => {
      wrapper.instance().forceUpdate = jest.fn()
      wrapper.instance().changeRoute()
      expect(wrapper.instance().forceUpdate).toHaveBeenCalled()
    })
  })

  describe('logoutUser', () => {
    it('should invoke signOut', () => {
      wrapper.instance().logoutUser()
      expect(wrapper.instance().props.signOut).toHaveBeenCalledWith({})
    })
  })

  describe('mapStateToProps', () => {
    it('should create the correct props object', () => {
      let mockState = {
        user: {id: 2, user: 'Bob', email: "bob@bob.com", loggedIn: true}
      }
      const  expected = {
        user: {id: 2, user: 'Bob', email: 'bob@bob.com', loggedIn: true}
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)      
    })
  })

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn()

    it('should map a key of signOut', () => {
      const dispatchedProps = mapDispatchToProps(mockDispatch)
      expect(dispatchedProps.signOut).toBeDefined()
    })

    it('should have signOut call dispatch', () => {
      const mockUser = {id: 2, user: 'Bob', email: 'bob@bob.com', loggedIn: true}
      const dispatchedProps = mapDispatchToProps(mockDispatch)
      dispatchedProps.signOut(mockUser)
      expect(mockDispatch).toHaveBeenCalled()
    })
  })
});