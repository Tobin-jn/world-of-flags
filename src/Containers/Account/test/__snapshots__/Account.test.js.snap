// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account should render like the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Account />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="account-area"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="back-button"
            onClick={[Function]}
            to="/"
          />
          <div
            className="account-logout"
          >
            log out
          </div>
        </div>,
        <div
          className="profile-image-container"
        />,
        <div
          className="user-data"
        >
          <p
            className="profile-name"
          >
            Alex Ela
          </p>
          <div
            className="points-container"
          >
            <div
              className="points-label"
            >
              points
            </div>
            <div
              className="points-number"
            >
              25
            </div>
          </div>
          <div
            className="account-data"
          >
            <div
              className="email-label"
            >
              email
            </div>
            <div
              className="email-text"
            >
              email@email.com
            </div>
          </div>
          <div
            className="extra-data"
          />
        </div>,
      ],
      "className": "Account",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="back-button"
              onClick={[Function]}
              to="/"
            />,
            <div
              className="account-logout"
            >
              log out
            </div>,
          ],
          "className": "account-area",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "className": "back-button",
              "onClick": [Function],
              "to": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "log out",
              "className": "account-logout",
            },
            "ref": null,
            "rendered": "log out",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "profile-image-container",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="profile-name"
            >
              Alex Ela
            </p>,
            <div
              className="points-container"
            >
              <div
                className="points-label"
              >
                points
              </div>
              <div
                className="points-number"
              >
                25
              </div>
            </div>,
            <div
              className="account-data"
            >
              <div
                className="email-label"
              >
                email
              </div>
              <div
                className="email-text"
              >
                email@email.com
              </div>
            </div>,
            <div
              className="extra-data"
            />,
          ],
          "className": "user-data",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Alex Ela",
              "className": "profile-name",
            },
            "ref": null,
            "rendered": "Alex Ela",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="points-label"
                >
                  points
                </div>,
                <div
                  className="points-number"
                >
                  25
                </div>,
              ],
              "className": "points-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "points",
                  "className": "points-label",
                },
                "ref": null,
                "rendered": "points",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "25",
                  "className": "points-number",
                },
                "ref": null,
                "rendered": "25",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="email-label"
                >
                  email
                </div>,
                <div
                  className="email-text"
                >
                  email@email.com
                </div>,
              ],
              "className": "account-data",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "email",
                  "className": "email-label",
                },
                "ref": null,
                "rendered": "email",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "email@email.com",
                  "className": "email-text",
                },
                "ref": null,
                "rendered": "email@email.com",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "extra-data",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="account-area"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="back-button"
              onClick={[Function]}
              to="/"
            />
            <div
              className="account-logout"
            >
              log out
            </div>
          </div>,
          <div
            className="profile-image-container"
          />,
          <div
            className="user-data"
          >
            <p
              className="profile-name"
            >
              Alex Ela
            </p>
            <div
              className="points-container"
            >
              <div
                className="points-label"
              >
                points
              </div>
              <div
                className="points-number"
              >
                25
              </div>
            </div>
            <div
              className="account-data"
            >
              <div
                className="email-label"
              >
                email
              </div>
              <div
                className="email-text"
              >
                email@email.com
              </div>
            </div>
            <div
              className="extra-data"
            />
          </div>,
        ],
        "className": "Account",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="back-button"
                onClick={[Function]}
                to="/"
              />,
              <div
                className="account-logout"
              >
                log out
              </div>,
            ],
            "className": "account-area",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "className": "back-button",
                "onClick": [Function],
                "to": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "log out",
                "className": "account-logout",
              },
              "ref": null,
              "rendered": "log out",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "profile-image-container",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="profile-name"
              >
                Alex Ela
              </p>,
              <div
                className="points-container"
              >
                <div
                  className="points-label"
                >
                  points
                </div>
                <div
                  className="points-number"
                >
                  25
                </div>
              </div>,
              <div
                className="account-data"
              >
                <div
                  className="email-label"
                >
                  email
                </div>
                <div
                  className="email-text"
                >
                  email@email.com
                </div>
              </div>,
              <div
                className="extra-data"
              />,
            ],
            "className": "user-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Alex Ela",
                "className": "profile-name",
              },
              "ref": null,
              "rendered": "Alex Ela",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="points-label"
                  >
                    points
                  </div>,
                  <div
                    className="points-number"
                  >
                    25
                  </div>,
                ],
                "className": "points-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "points",
                    "className": "points-label",
                  },
                  "ref": null,
                  "rendered": "points",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "25",
                    "className": "points-number",
                  },
                  "ref": null,
                  "rendered": "25",
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="email-label"
                  >
                    email
                  </div>,
                  <div
                    className="email-text"
                  >
                    email@email.com
                  </div>,
                ],
                "className": "account-data",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "email",
                    "className": "email-label",
                  },
                  "ref": null,
                  "rendered": "email",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "email@email.com",
                    "className": "email-text",
                  },
                  "ref": null,
                  "rendered": "email@email.com",
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "extra-data",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
